### PR TITLE
FileStore: potential memory leak if getattrs fails.

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/13637

### CODE REVIEW

File: src/os/FileStore.cc [hunk 1]
```diff
@@ -3780,6 +3780,7 @@ int FileStore::_fgetattrs(int fd, map<string,bufferptr>& aset)
     dout(10) << " -ERANGE, got " << len << dendl;
     if (len < 0) {
       assert(!m_filestore_fail_eio || len != -EIO);
+      delete[] names2;
```
Comment (ifed01) [inline src/os/FileStore.cc]: May be it's better to use std::unique_ptr to avoid the same issue if code changes?

**Comment by smithfarm** [comment]
@ifed01 Thanks for the review! Please note that this is a backport (to hammer) of ace7dd096b58a88e25ce16f011aed09269f2a2b4

**Comment by smithfarm** [comment]
@xiexingguo Please note the question at https://github.com/ceph/ceph/pull/6420#discussion_r45331686

**Comment by dachary** [comment]
@tchaikov does this backport look good to merge ? It passed a run of the hammer rados suite ( see http://tracker.ceph.com/issues/13356#note-28 for details ).

**Comment by tchaikov** [comment]
@dachary lgtm.

**Pull request merged**